src/groupeddataframe/groupeddataframe.jl

```diff
@@ -345,6 +360,8 @@ function Base.to_index(gd::GroupedDataFrame, idxs::AbstractVector{T}) where {T}
         Tuple
     elseif E1 <: NamedTuple
         NamedTuple
+    elseif E1 <: Dict
+        Dict
```

Comment:
This means we allow mixing `Dict`s with different signatures, but maybe it is OK? What do you think.
If we allow it please add test that something like: `[Dict(:a=>1), Dict("a"=>2)]` (appropriately adjusted) passes.

Reply:
But even if we keep it (i.e. allowing mixing) I would restrict it to `Union{Dict{Symbol}, Dict{<:AbstractString}}`. This will not change how the code works, but will produce a nicer error message `"Invalid index ...`. Now for invalid `Dict` you will get a `MethodError` when trying to call `to_index` and probably this will confuse users.

Reply:
I think it's better to be strict for now. I have added that code that will throw an error for different key types and added a test.